### DISCUSSION TITLE
Use Golang GitHub action in profile repository

### DISCUSSION THREAD

**Discussion by NickRTR** [community/community#41807] [Actions] [answered]
Hello developers,

I created a Go Github Action which works fine in the action's repository itself. I want to publish this Action to the GitHub Actions marketplace and use it in my personal profile repository. However, using the action in my profile repo results in the following error:

```
Run NickRTR/WakaTime-Readme@v0.3.6
Run go run ./
go: cannot find main module, but found .git/config in /home/runner/work/NickRTR/NickRTR
    to create a module there, run:
    go mod init
Error: Process completed with exit code 1.
```

I guess the error is because the action tries to run in the profile repository's files because it says there's only a README.md file. How do I fix this?

Action Repository: https://github.com/NickRTR/WakaTime-Readme

**Comment by airtower-luna** [comment, marked as answer]
Most likely the best option is to use a [Docker action](https://docs.github.com/en/actions/creating-actions/creating-a-docker-container-action) instead of a composite action. That way you can set up your Go tool in the Dockerfile, and then call it (as the entrypoint) with the workspace mounted as the working directory.

**Reply by NickRTR** [reply]
I wanted to do it without Docker originally but I guess that will be easier then. Thanks for your advice.

Anyways, if someone knows how to do it without Docker, please let me know:)

**Reply by airtower-luna** [reply]
Technically you can use [`github.action_path`](https://docs.github.com/en/actions/learn-github-actions/contexts#github-context) to find the code from your action repository. But then you'll have to deal with calling the code from there on the README in the workspace, and you'll depend on the Go version installed on the runner. Certainly doable, but more complex and less reliable. :wink:

**Reply by NickRTR** [reply]
Ok, then I'll try it with Docker, thanks!
